test/elbo.jl

```diff
@@ -22,7 +22,29 @@
     rng = MersenneTwister(123456)
     q_f = Normal.(zeros(10), ones(10))
 
-    @testset "$lik" for lik in Base.uniontypes(ApproximateGPs.ScalarLikelihood)
+    analytic_likelihoods = [
+        ExponentialLikelihood, GammaLikelihood, PoissonLikelihood, GaussianLikelihood
+    ]
+
+    @testset "testing all analytic implementations" begin
+        ### Test that we're not missing any analytic implementation in `analytic_likelihoods`!
+        default_quadrature_method_types = Set([
+            m.sig.types[2] for m in methods(ApproximateGPs._default_quadrature)
+        ])
+        delete!(default_quadrature_method_types, Any)  # ignore fallback
+        for lik in default_quadrature_method_types
+            if lik.body.name.name == :GammaLikelihood
+                # workaround while waiting for JuliaGaussianProcesses/GPLikelihoods.jl#41
+                @test any(
+                    l.body.name.name == lik.body.name.name for l in analytic_likelihoods
+                )
+            else
+                @test any(lik <: l for l in analytic_likelihoods)
+            end
+        end
+    end
+
+    @testset "$lik" for lik in analytic_likelihoods
```

Comment:
The test below isn't necessarily just for analytic likelihoods - it also just tests Gauss-Hermite vs MC for non-analytic, so maybe we do just need to keep a manually updated list of all likelihoods we want to test?

Reply:
What value do you see in separately testing MC vs GH for different likelihoods? I was thinking a single test comparing the two would be enough to satisfy they're both valid implementations of the expectation - then the actual function of which to take the expectation no longer matters...

Reply:
I mean, we can add more likelihoods to test anyways, and rename it to `likelihoods_to_test` if you want. (Whilst keeping the test that for all likelihoods for which there is a `_default_quadrature` implementation, make sure we explicitly test them.)

Reply:
I suppose it could be valuable if someone tries to add a more general (non-scalar) likelihood? GH/MC might not throw an error but could still give the wrong answer

Reply:
that's true! i've renamed it.